### PR TITLE
feat(alexaSkill): add alexaSkill event type

### LINKED ISSUE #17: Support Alexa Skills Kit events
(Also needs implementation in other languages!)

Documented here: https://developer.amazon.com/docs/custom-skills/request-and-response-json-reference.html

Example event:
```
{
  "version": "1.0",
  "session": {
    "new": true,
    "sessionId": "amzn1.echo-api.session.[unique-value-here]",
    "application": {
      "applicationId": "amzn1.ask.skill.[unique-value-here]"
    },
    "attributes": {
      "key": "string value"
    },
    "user": {
      "userId": "amzn1.ask.account.[unique-value-here]",
      "accessToken": "Atza|AAAAAAAA...",
      "permissions": {
        "consentToken": "ZZZZZZZ..."
      }
    }
  },
  "context": {
    "System": {
      "device": {
        "deviceId": "string",
        "supportedInterfaces": {
          "AudioPlayer": {}
        }
      },
      "application": {
        "applicationId": "amzn1.ask.skill.[unique-value-here]"
      },
      "user": {
        "userId": "amzn1.ask.account.[unique-value-here]",
        "accessToken": "Atza|AAAAAAAA...",
        "permissions": {
          "consentToken": "ZZZZZZZ..."
        }
      },
      "apiEndpoint": "https://api.amazonalexa.com",
      "apiAccessToken": "AxThk..."
    },
    "AudioPlayer": {
      "playerActivity": "PLAYING",
      "token": "audioplayer-token",
      "offsetInMilliseconds": 0
    }
  },
  "request": {}
}
```

### PR DESCRIPTION
**Pull request by coreylight**
- Adds support for Alexa Intent, and Alexa Start/End Session event types.
- Adds `idx` dev dependency for terse yet performant event-type check
- Adds `flat` dep for capturing unanticipated object key/values
- A few style changes, push least-anticipated event types further in the event check

fix #17

### CODE REVIEW

File: src/plugins/alexaSkill.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    idx(e, o => o.session.user) &&
+    idx(e, o => o.context.System) &&
+    idx(e, o => o.request.requestId)
+    ? type
+    : false;
+}
+
+function plugin(event, log) {
+  const obj = flatten(event);
+  Object.keys(obj).forEach(key => {
```
Comment (ewindisch) [inline src/plugins/alexaSkill.js]: so we're logging all of the keys?

Comment (ewindisch) [reply src/plugins/alexaSkill.js]: It looks like keys we *don't* want capture (at least):

```@iopipe/event-info.alexaSkill.context.System.apiAccessToken
@iopipe/event-info.alexaSkill.context.System.user.accessToken
@iopipe/event-info.alexaSkill.context.System.user.permissions.consentToken
```

Comment (coreylight) [reply src/plugins/alexaSkill.js]: I can blacklist those, thanks @ewindisch makes it easy to update!

**Comment by coreylight** [comment]
@ewindisch @kolanos back in review, added blacklisted keys requested by @ewindisch

**Comment by kolanos** [comment]
@coreylight @ewindisch Made equivalent updates here: https://github.com/iopipe/iopipe-python/pull/218

**Pull request merged**